engine/api/repository/dao_vcs_project_repository_analyze.go

```diff
@@ -0,0 +1,128 @@
+package repository
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-gorp/gorp"
+	"github.com/rockbears/log"
+
+	"github.com/ovh/cds/engine/api/database/gorpmapping"
+	"github.com/ovh/cds/engine/gorpmapper"
+	"github.com/ovh/cds/sdk"
+)
+
+func InsertAnalyze(ctx context.Context, db gorpmapper.SqlExecutorWithTx, analyze *sdk.ProjectRepositoryAnalyze) error {
+	// Count number of analyze
+	nb, err := countAnalyzeByRepo(db, analyze.ProjectRepositoryID)
+	if err != nil {
+		return err
+	}
+	if nb >= 50 {
```

Comment:
It's weird to delete when we insert.
I think i would prefer a goroutine that clean the analyze regulary, like we do for all other tables

Reply:
done